sdk/eventhub/eventhubs-checkpointstore-blob/samples-dev/receiveEventsUsingCheckpointStore.ts

```diff
@@ -13,22 +13,22 @@
 import { DefaultAzureCredential } from "@azure/identity";
 import { EventHubConsumerClient } from "@azure/event-hubs";
 import { BlobCheckpointStore } from "@azure/eventhubs-checkpointstore-blob";
-import { ContainerClient } from "@azure/storage-blob";
+import { BlobServiceClient } from "@azure/storage-blob";
 import "dotenv/config";
 
 const fullyQualifiedNamespace = process.env["EVENTHUB_FQDN"] || "<fully qualified namespace>";
 const eventHubName = process.env["EVENTHUB_NAME"] || "<eventHubName>";
 const consumerGroup =
   process.env["EVENTHUB_CONSUMER_GROUP"] || EventHubConsumerClient.defaultConsumerGroupName;
-const storageContainerUrl =
-  process.env["STORAGE_CONTAINER_URL"] ||
-  "https://<storageaccount>.blob.core.windows.net/<containername>";
+const storageEndpoint =
+  process.env["STORAGE_ENDPOINT"] || "https://<storageaccount>.blob.core.windows.net";
 
 export async function main() {
   const credential = new DefaultAzureCredential();
   // This client will be used by our eventhubs-checkpointstore-blob, which
   // persists any checkpoints from this session in Azure Storage
-  const containerClient = new ContainerClient(storageContainerUrl, credential);
+  const storageClient = new BlobServiceClient(storageEndpoint, credential);
+  const containerClient = storageClient.getContainerClient("checkpointstore");
 
```

Comment:
nit: you can still create container client directly

```suggestion
const containClient = new ContainerClient(`${storageEndpoint}/checkpointstore`, credential);
```

Reply:
right, but I am a bit warry of messing with URLs and their encoding and all that.

Reply:
Either way is fine for this sample.  From storage point of view, it is common that only admins have service level permission, and standard users only have permissions to a container.

Reply:
good point, however, that shouldn't be an issue in this code because creating a client doesn't by itself require the elevated permissions, right? I understand there could be confusion that the user may use the blob service client to do restricted things and I am open to ideas on how to address it.

Reply:
> that shouldn't be an issue in this code because creating a client doesn't by itself require the elevated permissions, right?

right. creating client is local operation.